ika/tests/test_shaker_driver.py

```diff
@@ -71,23 +63,37 @@ async def test_setpoint_roundtrip():
     """Confirm that the setpoint can be updated."""
     async def get():
         async with Shaker(ADDRESS) as device:
-            with pytest.raises(ValueError):
-                await device.set_speed(setpoint=-1)
-            with pytest.raises(ValueError):
-                await device.set_speed(setpoint=3001)
-            speed_sp = round(uniform(15, 1000), 0)
-            await device.set_speed(setpoint=speed_sp)
-            assert speed_sp == await device.get_speed()
+            speed_sp = round(uniform(100, 1000), 0)
+            temp_sp = round(uniform(30, 150), 2)
+            await device.set(equipment='shaker', setpoint=speed_sp)
+            await device.set(equipment='heater', setpoint=temp_sp)
+            response = await device.get()
+            assert speed_sp == response['speed']['setpoint']
+            assert temp_sp == response['temp']['setpoint']
     await get()
 
 
 async def test_start_stop():
     """Confirm that the shaker motor can be controlled."""
     async def get():
         async with Shaker(ADDRESS) as device:
-            assert await device.get_running() is False
-            await device.control(on=True)
-            assert await device.get_running() is True
-            await device.control(on=False)
-            assert await device.get_running() is False
+            response = await device.get()
```

Comment:
The point of having separate tests here is to test each function.  There is already a test for `get()`, so this test should be confined to `get_running()`.  (This isn't always possible, but is in this case)

Reply:
<img width="365" alt="Screen Shot 2023-04-10 at 2 42 56 PM" src="https://user-images.githubusercontent.com/67432509/230983791-820f51a3-b7fb-42ae-bc30-eeeaabc12c19.png">
There is no `get_running()`, so I'm confused. The function being tested here is `control()` not `get()`. This is consistent with the drivers you've written.
https://github.com/numat/ika/blob/0b9036303ea75786b3ef86c220c1a4a4f7f30924/ika/tests/test_hotplate_driver.py#L76
https://github.com/numat/ika/blob/0b9036303ea75786b3ef86c220c1a4a4f7f30924/ika/tests/test_stir_driver.py#L84

Reply:
Oh, fair enough.  I got smarter on the `ismatec` driver but never went back and updated `ika`
https://github.com/numat/ismatec/blob/fd98ee74cbf9de71af28a5c21c4c80bed1d5f865/ismatec/tests/test_driver.py#L70